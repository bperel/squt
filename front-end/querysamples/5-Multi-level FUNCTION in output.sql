SELECT co.ContractID AS ID, CONCAT(SUBSTRING(p.firstName,0,1),'.', p.lastName) AS PersonDisplayedName, IF(p.lastName="Doe", "Classic", "Unusual") AS IsClassicLastName
FROM Contract co
INNER JOIN Person p ON co.personInChargeID = p.ID
WHERE co.isActive = 1;