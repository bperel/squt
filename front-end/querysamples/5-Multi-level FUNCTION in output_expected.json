{
   "Functions" : {
      "IsClassicLastName" : {
         "to" : "OUTPUT",
         "Constants" : {
            "Classic" : "Classic",
            "Unusual" : "Unusual"
         },
         "name" : "if",
         "alias" : "IsClassicLastName"
      },
      "PersonDisplayedName" : {
         "to" : "OUTPUT",
         "Constants" : {
            "." : "."
         },
         "name" : "concat",
         "alias" : "PersonDisplayedName"
      },
      "1" : {
         "to" : "PersonDisplayedName",
         "Constants" : {
            "1" : "1",
            "0" : "0"
         },
         "name" : "substr",
         "alias" : "1"
      },
      "3" : {
         "to" : "IsClassicLastName",
         "Constants" : {
            "Doe" : "Doe"
         },
         "name" : "=",
         "alias" : "3"
      }
   },
   "Tables" : {
      "contract" : {
         "co" : {
            "OUTPUT" : {
               "ContractID" : {
                  "-1" : "ID"
               }
            },
            "CONDITION" : {
               "personInChargeID" : {
                  "p.ID" : "JOIN_TYPE_STRAIGHT"
               },
               "isActive" : {
                  "1" : "JOIN_TYPE_STRAIGHT"
               }
            }
         }
      },
      "person" : {
         "p" : {
            "OUTPUT" : {
               "firstName" : {
                  "1" : "firstName"
               },
               "lastName" : {
                  "PersonDisplayedName" : "lastName",
                  "3" : "lastName"
               }
            }
         }
      }
   }
}